src/libs/ReportUtils.ts

```diff
@@ -216,294 +488,218 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
+    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    if ((typeof report === 'object' && isEmptyObject(report)) || report?.isWaitingOnBankAccount) {
         return false;
     }
 
     // In case the payment is scheduled and we are waiting for the payee to set up their wallet,
     // consider the report as paid as well.
-    if (report.isWaitingOnBankAccount && report.statusNum === CONST.REPORT.STATUS.APPROVED) {
+    if (report?.isWaitingOnBankAccount && report?.statusNum === CONST.REPORT.STATUS.APPROVED) {
         return true;
     }
 
-    return report.statusNum === CONST.REPORT.STATUS.REIMBURSED;
+    return report?.statusNum === CONST.REPORT.STATUS.REIMBURSED;
 }
 
 /**
  * Whether the current user is the submitter of the report
- *
- * @param {String} reportID
- * @returns {Boolean}
  */
-function isCurrentUserSubmitter(reportID) {
+function isCurrentUserSubmitter(reportID: string): boolean {
     if (!allReports) {
         return false;
     }
-    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] || {};
-    return report && report.ownerEmail === currentUserEmail;
+    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    return report?.ownerEmail === currentUserEmail;
 }
 
 /**
  * Whether the provided report is an Admin room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isAdminRoom(report) {
+function isAdminRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.POLICY_ADMINS;
 }
 
 /**
  * Whether the provided report is an Admin-only posting room
- * @param {Object} report
- * @param {String} report.writeCapability
- * @returns {Boolean}
  */
-function isAdminsOnlyPostingRoom(report) {
-    return lodashGet(report, 'writeCapability', CONST.REPORT.WRITE_CAPABILITIES.ALL) === CONST.REPORT.WRITE_CAPABILITIES.ADMINS;
+function isAdminsOnlyPostingRoom(report: OnyxEntry<Report>): boolean {
+    return report?.writeCapability === CONST.REPORT.WRITE_CAPABILITIES.ADMINS;
 }
 
 /**
  * Whether the provided report is a Announce room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isAnnounceRoom(report) {
+function isAnnounceRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE;
 }
 
 /**
  * Whether the provided report is a default room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isDefaultRoom(report) {
-    return [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL].indexOf(getChatType(report)) > -1;
+function isDefaultRoom(report: OnyxEntry<Report>): boolean {
+    return [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL].some((type) => type === getChatType(report));
```

Comment:
```suggestion
    return [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL].includes(getChatType(report));
```
This is more correct ig

Reply:
Actually the output is the same , but I decided to use `some` because using includes I would need to add assertion to getChatType